security/oc_tls.c

```diff
@@ -2449,6 +2494,18 @@ oc_tls_init_connection(oc_message_t *message)
 }
 #endif /* OC_CLIENT */
 
+size_t
+oc_tls_send_message(oc_message_t *message)
+{
+#if defined(OC_CLIENT)
+  if (!oc_tls_connected(&message->endpoint)) {
+    oc_tls_init_connection(message);
```

Comment:
What if we get multiple calls to `oc_tls_send_message` with a peer that hasn't finished the TLS handshake yet. We will have multiple calls to `oc_tls_init_connection` -> multiple calls to `oc_tls_handshake` -> multiple calls to `mbedtls_ssl_handshake` with the same ssl context. Is that OK for mbedTLS?

Reply:
This is same behavior as was before. But as I known I expect that mbedtls has own state machine to control such thing, but I added check that the handshake is called just once.